test/src/commands/packages/commands/check/commands/licenses_test.dart

```diff
@@ -744,6 +783,50 @@ void main() {
           }),
         );
 
+        test(
+          'when a single license is not allowed and forbidden is left empty',
+          withRunner(
+              (commandRunner, logger, pubUpdater, pubLicense, printLogs) async {
+            final tempDirectory = Directory.systemTemp.createTempSync();
+            addTearDown(() => tempDirectory.deleteSync(recursive: true));
+
+            File(path.join(tempDirectory.path, pubspecLockBasename))
+                .writeAsStringSync(_validMultiplePubspecLockContent);
+
+            when(() => logger.progress(any())).thenReturn(progress);
+
+            const dependency1Name = 'very_good_test_runner';
+            when(() => pubLicense.getLicense(dependency1Name))
+                .thenAnswer((_) => Future.value({'MIT'}));
+            final license1LinkedMessage = link(
+              uri: pubLicenseUri(dependency1Name),
+              message: 'MIT',
+            );
+
+            const dependency2Name = 'cli_completion';
+            when(() => pubLicense.getLicense(dependency2Name))
+                .thenAnswer((_) => Future.value({'BSD'}));
+
+            await commandRunner.run(
+              [
+                ...commandArguments,
+                allowedArgument,
+                'BSD',
+                forbiddenArgument,
+                '',
+                tempDirectory.path,
+              ],
+            );
+
+            final errorMessage =
+                '''1 dependency has a banned license: $dependency1Name ($license1LinkedMessage).''';
+
+            verify(
+              () => logger.err(errorMessage),
+            ).called(1);
+          }),
```

Comment:
Nit: I feel this would benefit from a trailing comma

Reply:
I personally don't like the formatting after the trailing comma (I find it less legible). I'm going to resolve these Nits, hopefully in the future the formatter solves this automatically by forcing an opinion.

```dart
// After the trailing comma
verify(
  () => logger.err(
    errorMessage,
  ),
).called(1);

// Before the trailing comma
verify(
  () => logger.err(errorMessage),
).called(1);
```